scripts/extensions/ai-widget/src/translations/translations-body.tsx

```diff
@@ -124,26 +124,21 @@ export default class TranslationsBody extends React.Component<IProps, IState> {
                     {this.props.mode === 'current' && (
                         <Button
                             onClick={() => {
-                                const currentDeskId = superdesk.entities.desk.getActiveDeskId();
-                                const taskData = (() => {
-                                    if (currentDeskId != null) {
-                                        const currentDesk = superdesk.entities.desk.getDeskById(currentDeskId);
+                                const {translate, get, patch} = superdesk.entities.article;
 
-                                        return {
-                                            user: article.task.user,
-                                            desk: currentDesk._id,
-                                            stage: currentDesk.working_stage,
-                                        };
-                                    }
-
-                                    return {user: article.task.user};
-                                })();
-
-                                superdesk.entities.article.createNewWithData({
-                                    body_html: translation,
-                                    task: taskData,
-                                    language: this.props.activeLanguageId,
-                                }, article.profile);
+                                translate(
+                                    article,
+                                    this.props.activeLanguageId,
+                                ).then((translatedItem) =>
+                                    get(translatedItem._id).then((fullTranslatedItem) => {
```

Comment:
Reduce the destructuring a bit so it's not `get(translatedItem._id)`,
but `article.get(translatedItem._id)` or `articleApi.get(translatedItem._id)`

It's getting hard to follow where these short generic methods(get,patch) are coming from.